packages/insomnia-app/app/network/axios-request.ts

```diff
@@ -25,7 +26,9 @@ export async function axiosRequest(config: AxiosRequestConfig) {
     }),
   };
 
-  if (proxyUrl) {
+  if (settings.proxyEnabled && urlInNoProxy(finalConfig.url, settings.noProxy)) {
+    finalConfig.proxy = false;
+  } else if (settings.proxyEnabled && proxyUrl) {
```

Comment:
```suggestion
  if (settings.proxyEnabled && proxyUrl && !urlInNoProxy(finalConfig.url, settings.noProxy)) {
```
this line is a no-op if I'm not mistaken since a false value is the same as undefined `finalConfig.proxy = false;` so we don't need the if else

Reply:
Yeah this is a weird one.  I beleive that if the proxy key is false axios won't use a proxy, and if it is undefined axios will use environmental proxy values.  So here, the edge case where the user has the proxy enabled if their is a match in the noProxy then axios would use environmental values.  Which might be confusing if the user was expecting a proxy not to be used for a specific domain but environmental values were picked up and used.

Reply:
I beleive that this is the line from axios in question.  Axios does different things if the key is false as compared to undefined.  https://github.com/axios/axios/blob/c5fe05bdff82e0fa4ace9a832e75052e1ee297f0/lib/adapters/http.js#L154

Reply:
my mistake, although it is still a little confusing to read the logic because of the No in the `urlInNoProxy` name, perhaps `isURLMatchedInNoProxyRule` is more descriptive.

Reply:
Good Idea!  Thank you, I like working with people who think about code readability.

Reply:
> So here, the edge case where the user has the proxy enabled if their is a match in the noProxy then axios would use environmental values

would HTTP_PROXY and HTTPS_PROXY also be respected by this axios request? I feel like we should limit this code to respect all or none of the env variable proxy overrides, as they are easy to forget since they are implicit none is my preference.

> Good Idea!

Thanks for your contribution!

Reply:
I'm glad you asked this, it made me stop and think it over.  I had made some UX assumptions that I don't think were right.

Short story: 
I modified the code to work how the rest of the code in the project handles the proxy settings UX.

Long story:
I spent a little time to investigate how the rest client code in the project handles the proxy settings by trying the below settings and then looking at the timeline to see if the proxy was used.  Here is what I found.

- example of url used: <https://git.acme.com/api/v3/repositories>
- example of no_proxy uri used: git.acme.com
- restarted insomnia after each test

| insomnia proxy setting | has uri in insomnia no Proxy settings | has uri in system no_proxy | insomnia used proxy |
| ---                    | ---                                   | ---                        | ---                 |
| enabled                | true                                  | false                      | false               |
| enabled                | true                                  | true                       | false               |
| enabled                | false                                 | false                      | true                |
| enabled                | false                                 | true                       | true                |
| disabled               | true                                  | false                      | false               |
| disabled               | true                                  | true                       | false               |
| disabled               | false                                 | false                      | false               |
| disabled               | false                                 | true                       | false               |

From the results on the right we can see that the HTTP_PROXY and HTTPS_PROXY env vars are never used by insomnia.  If the insomnis user config panel proxy settings are not enabled, a proxy is never used.  and therefore I think that the UX for the git-sync should follow that same paradigm.

In my expereince anything with proxies is potentially a little confusing so I threw some tests in for the proxy stuff to add some transparency of how the proxy settings are being used.  If anyone ever needs to work with this code in the future hopefully this helps them.  Though I'll admit the test names are too long.  If you have suggestions for better test names to make the intent of these tests more clear let me know.  I added the tests here: packages\insomnia-app\app\network\__tests__\axios-request.test.ts

Thanks,

Reply:
@jackkav a few questions

- I saw the emoji, is it safe to resolve this conversation chain?  
- How long does it typically take for something like this to make it into the product?  (I want to start using it, lol)

Thanks,

Reply:
Hi @JasonCubic yes I believe we have determined that the app should ignore system env vars when it comes to proxy settings so it keeps our code paths explicit and easier to reason about. I will continue to review this PR after the holidays. I also have a PR in progress consolidating other types of internal app requests to hit this code paths so there are a few more considerations to make before I can be confident in approving but this chain is resolved, thanks for your help clearing up the ambiguity in proxy resolution.
> How long does it typically take for something like this to make it into the product? (I want to start using it, lol)
You can start using this right away if you run with `npm run app-start` or package with `npm run app-package`
We are also working on a way to build releases more regularly from the develop so you can just download something once your PR is merged, although its still too early to tell how it will look.